install.sh

```diff
@@ -53,11 +57,46 @@ if [ -d ../operator ]; then
 
 	# pluck image tag off operator image 
 	tag=$(cat ../operator/kappnav.yaml | grep operator: | awk '{ split($0,p,":"); print p[3] }')
-
+	
 	echo Install kappnav to kubeenv $kubeenv
 	kubectl create namespace kappnav 
 
-	cat ../operator/kappnav.yaml | sed "s|kubeEnv: okd|kubeEnv: $kubeenv|" | sed "s|repository: kappnav/|repository: $org/kappnav-|" | sed "s|tag: $tag|tag: latest|" | sed "s|image: kappnav/operator:$tag|image: $org/kappnav-operator:latest|" | kubectl create -f - -n kappnav 
+	if [[ x$reposArg == 'x' || x$reposArg == "" ]]; then
+		cat ../operator/kappnav.yaml | sed "s|kubeEnv: okd|kubeEnv: $kubeenv|" | sed "s|repository: kappnav/|repository: $org/kappnav-|" | sed "s|tag: $tag|tag: latest|" | sed "s|image: kappnav/operator:$tag|image: $org/kappnav-operator:latest|" | kubectl create -f - -n kappnav 
+	else
+		# parse the arguments using "," deliminiter
+		IFS=',' read -r -a repos <<< "$reposArg"
+
+		org=$DOCKER_USER
+		echo $DOCKER_PWD | docker login docker.io -u $DOCKER_USER --password-stdin
+		. ./projectList.sh
+		projs=$IMAGES
+		
+		cat ../operator/kappnav.yaml | \
+    		sed "s|kubeEnv: okd|kubeEnv: $kubeenv|" | \
+    		sed "s|tag: $tag|tag: dev|" | \
+    		sed "s|image: kappnav/operator:$tag|image: kappnav/operator:dev|" \
+    		> temp-kappnav.yaml
+		for p in $projs; do
+    		for repo in "${repos[@]}"; do
+        		if [ x$repo == x$p ]; then
+					if [ x$repo == x"operator" ]; then
+						sed "s|image: kappnav/operator:dev|image: $DOCKER_USER/kappnav-operator:latest|" temp-kappnav.yaml > temp-kappnav-new.yaml
+					else
+            			r="repository: kappnav/"$repo
+						# get the line number of the repo that the tag need to be updated
+						ln=`grep -n "$r" ../operator/kappnav.yaml | awk -F: '{print $1}'`
+						newln=$(($ln+1))
+						cat temp-kappnav.yaml | \
+						sed "$ln s|repository: kappnav/|repository: $DOCKER_USER/kappnav-|" | \
+    					sed "$newln s|tag: dev|tag: latest|" > temp-kappnav-new.yaml
+					fi
+					cat temp-kappnav-new.yaml > temp-kappnav.yaml
+        		fi
+    		done
+		done
+		kubectl create -f temp-kappnav.yaml -n kappnav
```

Comment:
temp files should be cleaned up.

Reply:
I didn't clean it up as that is the file to proof what image/tag is it is using instead needed. Should I remove it anyway?

Reply:
removed